File: 30daysOFcode_Java/day 29.java
import java.util.*;
//most precised and simple code for this problem...
//simplicity of the code makes it extremely easy to understand...
public class Solution {
    public static void main(String[] args) {
        //object crreation  for memory allocation to variables...
        Scanner sc=new Scanner(System.in);
        int t = sc.nextInt();
        for(int i=1;i<=t;i++){
        int n,k,max=0;
        n=sc.nextInt();
        k=sc.nextInt();
            for(int j=1;j<n;j++)
            {
                for(int l=j+1;l<=n;l++){
                    //bitwise-and operation...
                    int temp=l&j;
                    if((temp<k) && (temp>max)) max=temp;
                }}
            System.out.println(max);
}}}

File: 30daysOFcode_Scala/day 29.scala
object Solution {

  def main(args: Array[String]) {
    val sc = new java.util.Scanner (System.in);
    var t = sc.nextInt();
    var a0 = 0;
    while(a0 < t){
      var n = sc.nextInt();
      var k = sc.nextInt();
      a0+=1;
      val list = List.range(1, n+1, 1)
      var max = 0
      for (i <- 1 to n-1) {
        for (j <- i+1 to n) {
//          println(s"a: ${i} b: ${j}: (a&b): ${(i & j)}")
          if ((i & j) > max && (i & j) < k) {

            max = (i & j)
          }
        }
      }
      println(max)
    }
  }
}
